Utils/src/SignalScanProducer.cc

```diff
@@ -186,20 +220,31 @@ void SignalScanProducer::getSUSYModelInfo(std::string comment){
 	//underscore-delimited data
 	parse::process(comment,'_',fields);
 
-	//several possible formats:
-	//model name_mMother_mLSP (1+2 fields)
-	//model name_xChi_mMother_mLSP (1+3 fields)
-	//model name_name_name_mMother_mLSP (3+2 fields)
-	//just take last two values and convert to doubles
+	if (type_ == signal_type::SUSYGenPart) {
```

Comment:
personally, I would prefer this to be in its own function, rather than being shoved into the middle of a function that's handling another case